Automate weekly helm development release (without docs)

dimitarvdimitrov opened the pull request:
#### What this PR does
Adds a github actions job that creates a PR which updates the helm chart. https://github.com/grafana/mimir/pull/2622 is a PR created via the action.

The action runs at 12:00 UTC on Mondays and can also be triggered manually.

krajorama commented on .github/workflows/helm-weekly-release-pr.yaml:
```diff
@@ -0,0 +1,30 @@
+
+on:
+  schedule:
+    - cron: '0 12 * * 1' # 12 UTC on Mondays for weekly release; images for GEM and Mimir should have already been pushed
```
Needs to run on workdays if we missed the image generation on Monday. Must be idempotent since it may be run by manually.

```suggestion
    - cron: '0 12 * * 1-5' # 12 UTC on workdays for weekly release; images for GEM and Mimir should have already been pushed
```

krajorama commented on .github/workflows/scripts/helm-weekly-release.sh:
```diff
@@ -0,0 +1,68 @@
+update_yaml_node $values_file .image.tag $latest_mimir_tag
+update_yaml_node $values_file .smoke_test.image.tag $latest_mimir_tag
+update_yaml_node $values_file .enterprise.image.tag $latest_gem_tag
+update_yaml_node $chart_file .appVersion $latest_mimir_tag
```
There is no documentation for weekly releases and changing this makes everything restart (as the checksum of the config includes the whole configmap object. Let's remove this from the change.
```suggestion
```

krajorama commented on .github/workflows/scripts/helm-weekly-release.sh:
```diff
@@ -0,0 +1,68 @@
+}
+
+latest_mimir_tag=$(find_latest_image_tag grafana/mimir)
+latest_gem_tag=$(find_latest_image_tag grafana/enterprise-metrics)
```
We should check that both are on the same r* version

krajorama commented on .github/workflows/scripts/helm-weekly-release.sh:
```diff
@@ -0,0 +1,68 @@
+    # _This_ weekly release should have a semver that's one above the
+    new_chart_semver=$(increment_semver $current_chart_semver 1)
+  fi
+  echo "$new_chart_semver-weekly.$new_chart_weekly"
```
We should skip the rest if the version does not change (images were not pushed)

krajorama commented on .github/workflows/helm-weekly-release-pr.yaml:
```diff
@@ -0,0 +1,30 @@
+        uses: peter-evans/create-pull-request@v4
+        with:
+          token: ${{ secrets.GH_BOT_ACCESS_TOKEN }}
+          title: Release mimir-distributed Helm chart ${{ steps.update.outputs.new_chart_version }}
```
does this skip PR creation if the PR already exists with the same name? open or closed

dimitarvdimitrov replied:
I will discover that

dimitarvdimitrov replied:
if there's a closed PR it opens a new one. It did that quite a few times while I was testing. If the PR is open, it force pushes - see https://github.com/grafana/mimir/pull/2622

krajorama commented on .github/workflows/scripts/helm-weekly-release.sh:
```diff
@@ -0,0 +1,68 @@
+update_yaml_node $chart_file .appVersion $latest_mimir_tag
+update_yaml_node $chart_file .version $new_chart_version
+
+make TTY='' build-helm-tests doc
```
Should not need this

```suggestion
```

dimitarvdimitrov replied:
we agreed to still bump the image version, so we need to rebuild the golden records and regenerate the readme, I think we need this

krajorama replied:
oh right, it's the build-helm-tests that shouldn't be needed
```suggestion
make TTY='' doc
```

osg-grafana commented:
Do the weekly builds have documentation?

dimitarvdimitrov commented:
> Do the weekly builds have documentation?

no. They are dev releases, similar in significance to the weekly GEM and Mimir builds, which also don't have documentation.

dimitarvdimitrov commented:
I addressed your feedback @krajorama , can you take another look? Also https://github.com/grafana/mimir/pull/2622 was updated with the latest doings of this PR

dimitarvdimitrov commented:
are there plans to automate the weekly development release with docs @osg-grafana ? I'm asking because of the title change.
